ompi/communicator/comm.c

```diff
@@ -1662,21 +1663,19 @@ int ompi_comm_get_rprocs ( ompi_communicator_t *local_comm,
         goto err_exit;
     }
 
-    rbuf = OBJ_NEW(opal_buffer_t);
+    PMIX_DATA_BUFFER_CREATE(rbuf);
     if (NULL == rbuf) {
         rc = OMPI_ERR_OUT_OF_RESOURCE;
         goto err_exit;
     }
 
-    if (OMPI_SUCCESS != (rc = opal_dss.load(rbuf, recvbuf, rlen))) {
-        goto err_exit;
-    }
+    PMIX_DATA_BUFFER_LOAD(rbuf, recvbuf, rlen);
 
     /* decode the names into a proc-list -- will never add a new proc
        as the result of this operation, so no need to get the newprocs
        list or call PML add_procs(). */
     rc = ompi_proc_unpack(rbuf, rsize, &rprocs, NULL, NULL);
-    OBJ_RELEASE(rbuf);
+    PMIX_DATA_BUFFER_RELEASE(rbuf);
```

Comment:
Does the rc need to be converted in these cases via `opal_pmix_convert_status()`